### DISCUSSION TITLE
Fieldset

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#309] [Component suggestions]
<img width="720" alt="visuele weergave van de fieldset component" src="https://github.com/user-attachments/assets/1b42f833-0ff3-432d-bc77-02b25dce1d7a">

## Naam

Fieldset

## Doel

Groep van gerelateerde formuliervelden.

## Gerelateerde componenten
[Date Input Group](https://github.com/orgs/nl-design-system/discussions/271), [Text Input Group](https://github.com/orgs/nl-design-system/discussions/296), [Radio Group](https://github.com/orgs/nl-design-system/discussions/222), [Checkbox Group](https://github.com/orgs/nl-design-system/discussions/223)

## Onderzoek

[`<fieldset>`: The Fieldset element - MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset)
Documentatie van Mozilla Developers Network over het `<fieldset>` HTML element.

[Fieldset component - GOV.UK](https://design-system.service.gov.uk/components/fieldset/)
Fieldset component zoals beschikbaar bij GOV.UK.

[Fieldset Github Discussion - GOV.UK](https://github.com/alphagov/govuk-design-system-backlog/issues/48)
In deze Discussion van GOV.UK wordt Fieldset component besproken.

[Using the fieldset and legend elements](https://accessibility.blog.gov.uk/2016/07/22/using-the-fieldset-and-legend-elements/)
In dit artikel beschrijf Léonie Watson de toepassing van Fieldset en Legend.

[Use Legend and Fieldset - Adrian Roselli](https://adrianroselli.com/2022/07/use-legend-and-fieldset.html)
In dit artikel beschrijft Adrian Roselli de toepassing van Fieldset en Legend. Hij beschrijft onder andere:
- Een oplossing voor de layout.
- Hoe de combinatie van <fieldset> / <legend> en  group / aria-labelledby wordt voorgelezen door schermlezers.

[Fieldsets, Legends and Screen Readers again](https://www.tpgi.com/fieldsets-legends-and-screen-readers-again/)
In dit artikel beschrijft Steve Faulkner de toepassing van Fieldset en Legend. Hij beschrijft onder andere:
- Hoe een implementatie wordt voorgelezen door schermlezers.
- Dat headings binnen de `legend` element kunnen worden geplaatst.
- Hoe je met ARIA attributes ook voor groepering en labeling kan zorgen wanneer het niet mogelijk is om native HTML elementen te gebruiken.

[Brief Note on Disclosures in Fieldsets - Adrian Roselli](https://adrianroselli.com/2024/09/brief-note-on-disclosures-in-fieldsets.html)
In dit artikel beschrijf Adrian Roselli het gebruik van 'disclosure widgets' (zoals bijvoorbeeld een [Toggletip](https://github.com/orgs/nl-design-system/discussions/298)) in combinatie met Fieldsets.

**Comment by jeffreylauwers** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-forms-field-set--docs)
- [Figma](https://www.figma.com/design/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?node-id=21379-6998)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/field-set)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-09-13 om 14 42 21](https://github.com/user-attachments/assets/0b3e9276-f49e-48fc-8e22-8616f9591e43)

With Description

![Scherm­afbeelding 2024-09-13 om 14 42 31](https://github.com/user-attachments/assets/2f3ba4b4-9f2e-4982-95d8-f52184c88da6)

With Error

![Scherm­afbeelding 2024-09-13 om 14 42 36](https://github.com/user-attachments/assets/cca95210-bf5f-49da-90fd-053ff3066b9e)

Radio group

![Scherm­afbeelding 2024-09-13 om 14 42 42](https://github.com/user-attachments/assets/3595da05-dafe-4773-8e61-7ed8819a5bed)

Radio group with error

![Scherm­afbeelding 2024-09-13 om 14 42 47](https://github.com/user-attachments/assets/5f3440f9-d8e8-4112-bc51-38ed76646a55)

Checkbox group

![Scherm­afbeelding 2024-09-13 om 14 42 55](https://github.com/user-attachments/assets/6fa58431-1feb-4f34-ad70-28a970ac9599)


Checkbox group with error

![Scherm­afbeelding 2024-09-13 om 14 43 01](https://github.com/user-attachments/assets/b5af14ac-c170-4ec0-95e6-e065904c2e4b)

---

### Notities

Uitgebreide documentatie aanwezig voor implementatie.

**Reply by VincentSmedinga** [reply]
We schrijven de naam van dit component als twee losse woorden: Field Set.

Zo zou het in een Engelse zin ook geschreven worden. De namen van HTML-elementen moffelen spaties weg, bijv. ook voor blockquote (block quote), colgroup (column group) en figcaption (figure caption), en dat is in fieldset dus ook gebeurd.

Ook inhoudelijk klopt het: een Field Set kan een verzameling Fields bevatten, zoals meerdere adresvelden. Overigens zijn zelfs geneste Field Sets mogelkijk, zoals een lijst Radios die samen met andere velden in een Field Set hoort. Als directe kinderen van een Form mogen dus ook zowel Fields als Field Sets bestaan.

Overigens noemen we Field dus ook niet Form Field – dan zou je hier ook Form Field Set moeten aanhouden en dan wordt het wel wat lang allemaal. We verwachten het concept ‘Field’ niet buiten formulieren nodig te hebben of accepteren dat we in dat geval een andere naam moeten vinden.

Zie ook [mijn reactie bij Form Field](https://github.com/orgs/nl-design-system/discussions/311#discussioncomment-10914596).

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/story/open-forms-fieldset--default)
- [Figma (Huidig)](https://www.figma.com/design/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?node-id=11980-69026&t=xDZUoeKoHRBzdDIG-4)
- [Github](https://github.com/nl-design-system/denhaag/tree/main/components/FormFieldSet)

---

### Screenshots / Varianten

**Storybook**
![Scherm­afbeelding 2024-09-16 om 12 01 48](https://github.com/user-attachments/assets/bf491eec-53e1-4746-9cc4-dc9f97f3c504)

**Figma (huidig)**

![Scherm­afbeelding 2024-09-16 om 12 02 47](https://github.com/user-attachments/assets/4603ec32-32bf-4933-a512-eb5bab80e519)


---

### Notities

- Lijkt gebruik te maken van Fieldset van OpenFormulieren (zie Storybook)
- Documentatie aanwezig in Figma

**Comment by jeffreylauwers** [comment]
## Open Formulieren

[Storybook](https://open-formulieren.github.io/open-forms-sdk/?path=/docs/form-io-components-vanilla-fieldset--docs)

---

### Screenshots / Varianten

**Storybook**

![Scherm­afbeelding 2024-09-16 om 12 13 17](https://github.com/user-attachments/assets/332c8368-801a-4aaa-accc-663eb82df343)

---

### Notities

- Documentatie aanwezig in Storybook.
- Property beschikbaar om de legend te verbergen.
- Lijkt door [Den Haag](https://github.com/orgs/nl-design-system/discussions/309#discussioncomment-10657722) gebruikt te worden.

**Comment by jeffreylauwers** [comment]
## OpenGemeenten

[Patternlab](https://patternlab.lenteveld.nl/patterns/01-molecules-080-Forms-And-Search-Fieldset-Fieldset/01-molecules-080-Forms-And-Search-Fieldset-Fieldset.html)

---

### Screenshots / Varianten

**Patternlab**

Fieldset
![Scherm­afbeelding 2024-09-16 om 12 19 21](https://github.com/user-attachments/assets/d721a40e-8dc7-46a6-96a8-0b27b2d9cb2b)

Fieldset NoLabel
![Scherm­afbeelding 2024-09-16 om 12 19 26](https://github.com/user-attachments/assets/5fbb88fb-07f1-41f4-adb5-91858c34d591)

---

### Notities

Variant beschikbaar om de legend te verbergen.

**Comment by jeffreylauwers** [comment]
## RVO

- [Documentatie website](https://nl-design-system.github.io/rvo/docs/components/form-fieldset/)
- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-form-fieldset--documentatie)
- [Figma](https://www.figma.com/design/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?node-id=476-23636)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/form-fieldset)

---

### Screenshots / Varianten

**Documentatie website**
![Scherm­afbeelding 2024-09-16 om 12 30 23](https://github.com/user-attachments/assets/55fcde3b-7cd9-4a91-89e3-e4f69696bdf9)

**Storybook**
![Scherm­afbeelding 2024-09-16 om 12 31 35](https://github.com/user-attachments/assets/aa47117c-14fc-4c00-81a8-8800607a13fd)

**Figma**
![Scherm­afbeelding 2024-09-16 om 12 30 52](https://github.com/user-attachments/assets/a0ada5bb-9c4f-4709-92a5-8d39c952945e)

---

### Notities

- Fieldset heeft een achtergrondkleur
- In Storybook is er een 'disabled' property beschikbaar.

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-form-fieldset--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/form-fieldset)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-09-16 om 12 37 04](https://github.com/user-attachments/assets/f2db4b26-9415-4bac-a14e-cfcd68c6be9e)

Without CSS
![Scherm­afbeelding 2024-09-16 om 12 37 10](https://github.com/user-attachments/assets/24fad590-0e39-4699-b491-d4902198d31b)

`<div>` as markup
![Scherm­afbeelding 2024-09-16 om 12 37 17](https://github.com/user-attachments/assets/167ef2c3-9722-437f-be72-753a65a8b9f2)

Disabled
Een veldengroep waar alle velden `disabled` zijn.
![Scherm­afbeelding 2024-09-16 om 12 37 26](https://github.com/user-attachments/assets/ed3303c7-2e1b-48fd-b0f3-0e16598afa84)

Section
![Scherm­afbeelding 2024-09-16 om 12 37 32](https://github.com/user-attachments/assets/74a27d24-421f-41db-9255-8eb383e36806)

Invalid
Een veldengroep die niet valide is. Gebruik deze voor veldengroepen de als geheel gevalideerd worden, bijvoorbeeld een radio group of een checkbox group. Gebruik `invalid` niet voor veldengroepen waar één los onderdeel invalide is, bijvoorbeeld: als één van de vragen nog niet beantwoord is.
![Scherm­afbeelding 2024-09-16 om 12 37 37](https://github.com/user-attachments/assets/1dae2db5-9ee7-4c96-bb4b-f6f54840ffc7)


---

### Notities

- Uitgebreide documentatie aawezig.
- Veel states uitgewerkt.

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Fieldset op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=fieldset)

**Comment by eslook** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Fieldset op het Candidate bord](https://github.com/orgs/nl-design-system/projects/80/views/1?filterQuery=Fieldset)
